gather_for_strain.snakefile

```diff
@@ -0,0 +1,120 @@
+import pandas as pd
+import os
+
+m = pd.read_csv("inputs/metadata.csv", header = 0)
+m = m[m['pa_in_reads'] == True] # filter the metadata to samples that had Pa reads 
+SRX = list(m['experiment_accession'])
+STRAIN = ['pao1', 'pa14']
+
+h = pd.read_csv("inputs/hogan_metadata.csv", header = 0)
+SPU = list(h['sample'])
+
+
+rule all: 
+    input:
+        expand("outputs/rnaseq_sourmash_gather/{srx}_gtdb_k31.csv", srx = SRX),
+        expand("outputs/rnaseq_sourmash_gather_spu/{spu}_gtdb_k31.csv", spu = SPU)
+    
+###########################################################################
+## Download databases 
+###########################################################################
+
+rule sourmash_download_human_sig:
+    output: "inputs/sourmash_dbs/GCF_000001405.39_GRCh38.p13_rna.sig"
+    resources:
+        mem_mb = 1000,
+    threads: 1
+    shell:'''
+    wget -O {output} https://osf.io/anj6b/download
+    '''
+
+rule sourmash_download_gtdb_database_k31:
+    output: "inputs/sourmash_dbs/gtdb-rs202.genomic.k31.zip" 
+    resources:
+        mem_mb = 1000,
+    threads: 1
+    shell:'''
+    wget -O {output} https://osf.io/94mzh/download
+    '''
+
+
+###########################################################################
+## Publicly available files
+###########################################################################
+
+rule rnaseq_sample_sourmash_sketch:
+    """
+    Because some of the samples analyzed here were not already in the compendia,
+    some of them were downloaded already, with the local fastq file located in
+    inputs/raw/sra/{srx}.fastq.gz. If that file does exists, this rule will
+    sketch from that file. If it does not exists, this rule will streaming download
+    the SRA accession and pipe it to the sketch command, that way the raw data 
+    doesn't have to be written to disk. Because of this rule operating based on
+    the fastq.gz files existing, this snakefile should not be run in parallel with
+    the add_sra_to_compendia.snakefile that is also in this directory.
+    """
+    output: "outputs/rnaseq_sourmash_sketch/{srx}.sig"
+    resources:
+        mem_mb = lambda wildcards, attempt: attempt * 2000 ,
+    threads: 1
```

Comment:
`threads` controls the number of threads that e.g. slurm will give to the rule when it's running on a cluster. When specified this way, it also becomes a python variable that can be called within e.g. the shell command. If the code is parallelizable with a flag (like `-t`), then you can write `-t {threads}` and it will scoop in the number of threads based on what's specified in the rule. Since this is also given to slurm by snakemake, it's works really nicely for code that is already parallelize-able. It would figure out how to parallelized code that doesn't know how to run in parallel. `sourmash sketch` can only run on one thread, so I only put a `1` here. 

The other handy thing snakemake will do is limit the total number of processes running at once. So when i go to run this on the command line, I would use `-j` to specify the maximum number of processes that can run at a time. so if i used `-j 16`, I'm telling snakemake that up to 16 threads can be started at a given time. So in this case, 16 sketch commands could launch because all of them only use 1 thread.